### ISSUE TITLE
Update deprecated Pydantic v2 methods

### ISSUE THREAD

**Issue by cruisen** [koxudaxi/datamodel-code-generator#2213]
The package currently uses several deprecated Pydantic v2 methods that should be updated to their new equivalents:

1. `parse_obj` -> `model_validate`
2. `dict` -> `model_dump`
3. `__fields_set__` -> `model_fields_set`

These deprecation warnings appear when using the package with Pydantic v2:

- PydanticDeprecatedSince20: The parse_obj method is deprecated; use model_validate instead. 
- PydanticDeprecatedSince20: The dict method is deprecated; use model_dump instead. 
- PydanticDeprecatedSince20: The __fields_set__ attribute is deprecated, use model_fields_set instead

These methods will be removed in Pydantic v3.0, so they should be updated to ensure future compatibility.

Relevant files:
- `datamodel_code_generator/parser/jsonschema.py`

See the [Pydantic V2 Migration Guide](https://docs.pydantic.dev/latest/migration/) for more details on the changes.

**Comment by koxudaxi** [comment]
Yes, this implementation is designed to support both v1 and v2. For backward compatibility reasons, we need to maintain support for both versions.
If this is causing issues for users, we could completely suppress these warnings. Would that be a better approach from a user perspective?

**Comment by riferg206** [comment]
We should be able to avoid suppressing warnings, which is preferable as both maintainer and user IMO. The deprecation warnings will still get thrown when the unit tests are run, but that may be unavoidable without adding version handling in the unit tests themselves. What is preferable from a maintainers standpoint? I'm new to open source, but if this seems sensible I can open a new PR with the changes from the one I closed and request to merge.
